### ISSUE TITLE
"Path is not relative" when GitVersionTask is used in ccproj files

### ISSUE THREAD

**Issue by AGBrown** [GitTools/GitVersion#1053]
Adding the Import

`<Import Project="..\packages\GitVersionTask.3.6.3\build\portable-net+sl+win+wpa+wp\GitVersionTask.targets" Condition="Exists('..\packages\GitVersionTask.3.6.3\build\portable-net+sl+win+wpa+wp\GitVersionTask.targets')" />`

into a ccproj (Azure cloud service) project results in the project not loading in Visual Studio with the error:

`path is not relative`

By a process of elimination, it is caused by [this block of ncrunch-related `ItemGroup` `None` definitions](https://github.com/GitTools/GitVersion/blob/v3.6.3/src/GitVersionTask/NugetAssets/GitVersionTask.targets#L82-L92). If those are commented out then the error does not occur. If the first one is left in, then the error comes back and the project will not load.

Workarounds:
- Comment out [this entire `ItemGroup`](https://github.com/GitTools/GitVersion/blob/v3.6.3/src/GitVersionTask/NugetAssets/GitVersionTask.targets#L81-L92)
- Add a `Condition=" $(NCrunch) != '' "` to the item group on [this line](https://github.com/GitTools/GitVersion/blob/v3.6.3/src/GitVersionTask/NugetAssets/GitVersionTask.targets#L82).

Are these ncrunch definitions required in the targets file once it is deployed to the nupkg? If not could they be removed, and if so could we just add the condition to them (I'm not sure if that condition is correct, but it "Works On My MachineTM")?

**Comment by asbjornu** [comment]
@AGBrown: We've had to do the same thing for NCrunch, so I assume that's a common solution, yes. What's causing it and whether there's another and possibly better solution available is perhaps something we should investigate if this continues to pop up. At worst, we should document the current behavior and how to avoid it.

**Comment by AGBrown** [comment]
@asbjornu - agreed

1. documentation would be useful (tbh I didn't work out the `Condition` workaround, I found it by searching the files in the GitVersion project for the workd `ncrunch` and noticing that a similar condition was used elsewhere. I suppose that's kind of like documentation - it didn't take me long to find it - and kudos to whoever worked it out in the first place).
1. Adding that condition to the GitVersionTask targets file would be great, but only if someone knows it won't break the reason for that ItemGroup in the first place (I'm not using ncrunch so can't verify that)

**Comment by asbjornu** [comment]
@AGBrown: If you're not using NCrunch, then why would you need to add `Condition=" $(NCrunch) != '' "` in the first place? 🤔

**Comment by AGBrown** [comment]
> If you're not using NCrunch, then why would you need to add Condition=" $(NCrunch) != '' " in the first place? :thinking:

TL;DR from the gitter discussion we had. This was a misunderstanding. The `Condition=" $(NCrunch) != '' "` is added to the [ItemGroup](https://github.com/GitTools/GitVersion/blob/v3.6.3/src/GitVersionTask/NugetAssets/GitVersionTask.targets#L82) only. It does not disable GitVersionTask. This [ItemGroupItemGroup](https://github.com/GitTools/GitVersion/blob/v3.6.3/src/GitVersionTask/NugetAssets/GitVersionTask.targets#L81-L92) was specifically added for NCrunch, and the condition removes it when NCrunch is not installed.

This condition is similar to the conditions [here](https://github.com/GitTools/GitVersion/blob/v3.6.3/src/GitVersionTask/GitVersionTask.csproj#L110) and [here](https://github.com/GitTools/GitVersion/blob/v3.6.3/src/GitVersionExe/GitVersionExe.csproj#L83) which also affect a section if NCrunch depending on the presence of NCrunch.

Summary:

1. That [entire ItemGroup](https://github.com/GitTools/GitVersion/blob/v3.6.3/src/GitVersionTask/NugetAssets/GitVersionTask.targets#L81-L92) seems to have been added to support NCrunch users
1. When NCrunch is not installed it doesn't affect csproj file usage (luckily)
1. But it does impact ccproj file usage - causing the cloud service projects not to load
1. Using GitVersionTask directly in ccproj files makes sense (rather than GitVersion.exe) as it is often built and packaged for testing from within Visual Studio.
